### PR TITLE
Command support

### PR DESCRIPTION
**Pull request by haras-unicorn**
Here is the [command branch](https://github.com/Hrle97/mapx.nvim/tree/cmd) on my fork. I was using a script for commands (and autocommands, but that's a bit harder to integrate, so expect a PR for that as well in the future if you like this one) and I was using this plugin and I saw that one of the goals was to create an API for commands and autocommands as well, so I figured that I could just contribute my scripts here instead of creating a plugin from scratch since both would fill the purpose of writing all of these things more ergonomically while the Neovim devs are figuring something out.

HOW IT WORKS:
The [``cmd.lua`` script](https://github.com/Hrle97/mapx.nvim/blob/cmd/lua/mapx/cmd.lua) exports itself as a function and that gets used in the ``mapx.lua`` script to add the appropriate function. I added a [test](https://github.com/Hrle97/mapx.nvim/blob/cmd/test/cmd.lua) and it seems to be working well. I also added a line in the globalize function so that it works as expected.

DOCS:
The exported function has docs that doesn't deviate a lot from the rest of the documentation (hopefully because there is a lot to document for that single function). I also added a chapter in the help file. The help seems a bit out of place there, but I hope it is a good start for now.

TODO:
The only thing I would like to do still, which I didn't, is support for command attributes like ``bang``. I know mapx supports analogous attributes for mappings by exporting singleton tables like ``expr``. The reason I didn't do this is because I think it would require a bit of an overhaul of the help file and I don't know if I would do that correctly because I am very new to writing help in Neovim.

POSSIBLE ISSUES:
Note that my English grammar is not the best, so there are bound to be some spelling/syntax errors in the docs/help. 
I added a ``.luacheckrc`` file for the ``luacheck`` linter which I use alongside the ``sumneko_lua`` LSP. I also write most things up to 80 characters. These things are a bit superficial IMO, but they tend to annoy some people, so if you want me to fix something regarding this it's not a problem.

EDIT:
TBH, I didn't think about this at all, but the command always gets added with a ``!`` and that may be undesireable for some people, so I might make a ``cmdbang`` function or do that in another PR.

EDIT2:
Forgot to update the README as well. I updated it in the same fashion as the help files.

### CODE REVIEW

**Comment by b0o** [comment]
Thank you for working on this!

> TBH, I didn't think about this at all, but the command always gets added with a ! and that may be undesireable for some people, so I might make a cmdbang function or do that in another PR.

I agree a `cmdbang` function would be good as that stays consistent with the map API. 

I took a cursory glance and it looks good so far. I will review more thoroughly when I have time in the next few days. And thanks for adding tests!

**Comment by haras-unicorn** [comment]
No problem!

Added the ``cmdbang`` function as well. Docs, test, help and README included.

File: lua/mapx.lua [hunk 1]
```diff
@@ -1,21 +1,24 @@
...
-    mapFuncs[n] = mapx[n]
+    funcs[m] = mapx[m]
+    funcs[n] = mapx[n]
   end
-  mapFuncs.mapbang = mapx.mapbang
-  mapFuncs.noremapbang = mapx.noremapbang
-  for k, v in pairs(mapFuncs) do
+  funcs.mapbang = mapx.mapbang
+  funcs.noremapbang = mapx.noremapbang
+  funcs.cmd = mapx.cmd
```
Comment (b0o) [inline lua/mapx.lua:20]: This appears to be missing `funcs.cmdbang = mapx.cmdbang`

**Review by b0o — CHANGES_REQUESTED** [review]
Other than the single issue, this looks good to me. Apologies for the delay, but thank you again for your contribution!

File: lua/mapx.lua (hunk 1 above)
Comment (haras-unicorn) [reply lua/mapx.lua:20]: Oh right!

Comment (haras-unicorn) [reply lua/mapx.lua:20]: Added it into tests as well and workd as expected.

**Comment by b0o** [comment]
Thank you!

**Pull request merged**